### PR TITLE
Refactor

### LINKED ISSUE #506: BAB not calculating beyond 1st level
It seems that when any creature or NPC is imported, the BAB for their class(es) doesn't calculate beyond 1st level despite how many levels or HD they have. The only fix I've found is going in and removing the classes and readding them from the compendium. And I know it wasn't always doing this, the issue seems relatively new. I've ran into the issue now with multiple statblocks, both imported from the statblock library module and directly copy/pasted from aonprd.

### PR DESCRIPTION
**Pull request by mkahvi**
- Refactors the codebase to use sourcedata instead of derived data.
  - This prevents a large quantity of extraneous data being stored on actor, especially the kind that prevent normal actor operation.
    - For example classes no longer get permanently stored hitDice value that blocks PF1 from calculating it correctly.
- Removes bunch of extraneous data being intentionally being attempted to be stored (which Foundry scraps anyway).
- Removes several values being attempted to be added that are derived data.
- This also removed a health update that was causing the current health to be weird.
- Fixes spellbook enabling.

This also adds basic .editorconfig to encourage at least minimally consistent style as it was really hard to do editing without massive inconsistent styling issues.

Also made object keys more uniform by removing a lot of string usage, but that shouldn't matter for most anything.

~~Note: This may introduce small problem with current health not being maxed, but that should be easy to fix by adding an update to set health to max around the time the conversion buff is created.~~ Edit: This is now fixed.

Fixes #506 and #511

### CODE REVIEW

**Comment by Lavaeolous** [comment]
This breaks sometimes, i'll investigate.

But in general: Thank you very much for all the work you put into this!

How to reproduce error:
1. Try to parse https://www.aonprd.com/NPCDisplay.aspx?ItemName=Savage%20Plant%20Sage
2. Gets stuck in line 4012 of sbcParsers.js with `game.pf1.documents.ItemPF.toConsumable(...).toObject is not a function`

**Comment by mkahvi** [comment]
I was meaning to test it more, too, with extra creatures. I was working mostly with elder air elemental. More varied testing is obviously needed before it's merged.

**Comment by Lavaeolous** [comment]
The error happens with spell-based consumables, e.g. potions and stuff. The system wants me to use `ItemSpellPF`, as `ItemPF.toConsumable()` is deprecated. But `ItemSpellPF` is not found in game.pf1.documents. Maybe i'm a version behind, i'll check.

**Comment by mkahvi** [comment]
It's in `CONFIG.Item.documentClasses.spell`

**Comment by mkahvi** [comment]
I'd like to point out that some errors in how this produces info might be caused by upstream bugs: https://gitlab.com/Furyspark/foundryvtt-pathfinder1/-/issues/1360

Spellbooks have for example slightly suspicious data with the per level slots, but it's functional at least.

**Comment by mkahvi** [comment]
[Sound Warrior](https://www.aonprd.com/NPCDisplay.aspx?ItemName=Sound%20Warrior) still breaks it but with different part.

**Comment by mkahvi** [comment]
Okay, things seem to work out with the stated statblocks. The remaining bugs are not related to this PR that I can tell.

**Comment by Lavaeolous** [comment]
I'll check this again today and will merge if no other issues arise!

**Review by Lavaeolous — APPROVED** [review]
Thanks alot again!

**Pull request merged**
